### PR TITLE
docs(builder): add note about v3.0 vs v2.0 plugins

### PR DESCRIPTION
**Pull request by protango**
**What's the problem this PR addresses?**

Related to: #2872.

Fixes confusion relating to usage of the plugin builder, as the current version is for building Yarn 3 plugins, but this is not obvious. This is compounded by the fact that the [Yarn 2.0 docs recommend usage of the builder](https://yarnpkg.com/advanced/plugin-tutorial/#gatsby-focus-wrapper) without mentioning that you must use v2.x of the builder to make plugins compatible with Yarn 2.

**How did you fix it?**

Added a note to the builder readme explaining that the current version is for building Yarn 3 plugins, and recommending usage of `@yarnpkg/builder@^2` if the user wants to build plugins for 2.x.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-builder/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 A CLI tool designed for creating, building, and managing complex plugins.
 
+This version of the builder is for creating plugins for Yarn 3.x. Yarn 3 plugins are not compatible with Yarn 2 installations, however Yarn 2 plugins are usually compatible with Yarn 3. If you wish to create plugins for Yarn 2, please use v2.x of the builder (`npm install @yarnpkg/builder@^2`).
```
Comment (merceyz) [inline packages/yarnpkg-builder/README.md:5]:
```suggestion
This version of the builder is for creating plugins for Yarn 3.x. Yarn 3 plugins are not compatible with Yarn 2 installations, however Yarn 2 plugins are usually compatible with Yarn 3. If you wish to create plugins for Yarn 2, please use v2.x of the builder (`yarn add @yarnpkg/builder@^2`).
```

**Comment by protango** [comment]
@merceyz Thanks for editing the changes. I'm trying to resolve the [failing check](https://github.com/yarnpkg/berry/pull/2882/checks?check_run_id=2560925170), however when I run `yarn version check --interactive` locally it does not produce any output, even after moving my changes to another branch and resetting `master` to upstream. Do you know how I might fix this?

**Comment by merceyz** [comment]
> I selected "undecided" on bumping the version as I'm not sure of the policy since it's only a docs change.

You can set it to `declined`, but if you want it to be published to npm set it to `patch`

**Comment by protango** [comment]
Ok, set to `patch` for the builder but `declined` the other projects

File: packages/yarnpkg-builder/README.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 A CLI tool designed for creating, building, and managing complex plugins.
 
+This version of the builder is for creating plugins for Yarn 3.x. Yarn 3 plugins are not compatible with Yarn 2 installations, however Yarn 2 plugins are usually compatible with Yarn 3. If you wish to create plugins for Yarn 2, please use v2.x of the builder (`yarn add @yarnpkg/builder@^2`).
```
Comment (paul-soporan) [inline packages/yarnpkg-builder/README.md:5]: We usually put pieces of advice in blockquotes:

```suggestion
> This version of the builder is for creating plugins for Yarn 3.x. Yarn 3 plugins are not compatible with Yarn 2 installations, however Yarn 2 plugins are usually compatible with Yarn 3. If you wish to create plugins for Yarn 2, please use v2.x of the builder (`yarn add @yarnpkg/builder@^2`).
```

**Review by paul-soporan — APPROVED** [review]

**Pull request merged**